crates/block-producer/src/block_builder/mod.rs

```diff
@@ -352,3 +359,38 @@ impl TelemetryInjectorExt for ProvenBlock {
         span.set_attribute("block.commitments.transaction", header.tx_hash());
     }
 }
+
+// BLOCK PROVER
+// ================================================================================================
+
+pub enum BlockProver {
+    Local(LocalBlockProver),
+    Remote(RemoteBlockProver),
+}
+
+impl BlockProver {
+    pub fn new_local(security_level: u32) -> Self {
+        info!(target: COMPONENT, "Using local block prover");
+        Self::Local(LocalBlockProver::new(security_level))
+    }
+
+    pub fn new_remote(endpoint: impl Into<String>) -> Self {
+        info!(target: COMPONENT, "Using remote block prover");
+        Self::Remote(RemoteBlockProver::new(endpoint))
+    }
```

Comment:
Similar comment here re specifying the security level and then making sure that generated proofs comply with it as in the other PR.